config/v1/tests/authentications.config.openshift.io/ExternalOIDCWithUIDAndExtraClaimMappings.yaml

```diff
@@ -0,0 +1,712 @@
+            claimMappings:
+              extra:
+              - key: ""
+                valueExpression: claims.foo
+      expectedError: "Invalid value: \"\": spec.oidcProviders[0].claimMappings.extra[0].key in body should be at least 1 chars long"
+    - name: Cannot set OIDC providers with extra claim where valueExpression is missing
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              extra:
+              - key: example.com/foo
+      expectedError: "spec.oidcProviders[0].claimMappings.extra[0].valueExpression: Required value"
+    - name: Cannot set OIDC providers with extra claim where valueExpression is an empty string
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              extra:
+              - key: example.com/foo
+                valueExpression: ""
+      expectedError: "Invalid value: \"\": spec.oidcProviders[0].claimMappings.extra[0].valueExpression in body should be at least 1 chars long"
+    - name: Cannot set OIDC providers with extra claim with duplicate keys
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              extra:
+              - key: example.com/foo
+                valueExpression: "claims.foo"
+              - key: example.com/foo
+                valueExpression: "claims.foo"
+      expectedError: "Duplicate value: map[string]interface {}{\"key\":\"example.com/foo\"}"
+    - name: Cannot set OIDC providers with extra claim with a key that is not a valid RFC 1123 DNS subdomain
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              extra:
+              - key: ex@mpl3.com/foo
+                valueExpression: "claims.foo"
+      expectedError: "the domain of the key must consist of only lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character"
+  onUpdate:
+    - name: Updating OIDC provider with a client that's not in the status
+      initial: &initConfig |
```

Comment:
Would prefer not to use anchors and just duplicate where we have used them. It's harder to read/review/understand test cases when they are split like this

Reply:
I can update this. This was just copied from the existing tests for this API.

Reply:
Updated in the latest push to no longer use anchors.